src/main/java/KilogramsServlet.java

```diff
@@ -0,0 +1,50 @@
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+@WebServlet("/kilograms")
+public class KilogramsServlet extends HttpServlet {
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String kg = request.getParameter("kg");
+        String g = request.getParameter("g");
+        String mg = request.getParameter("mg");
+
+        response.setCharacterEncoding("UTF-8");
+        response.setContentType("text/html");
+        PrintWriter writer = response.getWriter();
+
+        if ((kg != null && !kg.equals("") && g != null && !g.equals("") && mg != null && !mg.equals(""))
```

Comment:
To można było inaczej załatwić - wystarczy że tutaj wywołasz metodę która zliczy sobie ile parametrów zostało wpisanych. To znaczy - utworzysz sobie zmienną np `int count = 0` następnie sprawdzisz każdy parametr czy jest nie `null` oraz jest nie pusty i jeżeli się taki okaże to dodajesz 1 do `count`. Następnie zwrócisz z tej metody informację czy `count == 1` - jeżeli tak to użytkownik wprowadził dokładnie 1 parametr, jeżeli nie to znaczy że trzeba mu zwrócić informację żę wprwadził za dużo.

Reply:
Poprawione